packages/server/src/errors.ts

```diff
@@ -299,39 +159,60 @@ export function formatApolloErrors(
   //   flattenedErrors.push(error);
   // }
 
-  const enrichedErrors = errors.map((error) => enrichError(error, debug));
-  const makePrintable = (error: GraphQLFormattedError) => {
-    if (error instanceof Error) {
-      // Error defines its `message` and other fields as non-enumerable, meaning JSON.stringify does not print them.
-      const graphQLError = error as GraphQLFormattedError;
-      return {
-        message: graphQLError.message,
-        ...(graphQLError.locations && { locations: graphQLError.locations }),
-        ...(graphQLError.path && { path: graphQLError.path }),
-        ...(graphQLError.extensions && { extensions: graphQLError.extensions }),
-      };
-    }
-    return error;
-  };
-
-  if (!formatter) {
-    return enrichedErrors;
-  }
+  const { includeStackTracesInErrorResponses, errorCode } = options;
 
-  return enrichedErrors.map((error) => {
+  const formatError = options.formatError ?? ((error) => error);
+  return errors.map((error) => {
     try {
-      return makePrintable(formatter(error));
+      return formatError(enrichError(error), error);
     } catch (err) {
-      if (debug) {
-        // XXX: This cast is pretty sketchy, as other error types can be thrown!
-        return enrichError(err as Partial<GraphQLError>, debug);
+      if (includeStackTracesInErrorResponses) {
+        return enrichError(err);
```

Comment:
This seems like the same as before and good logic but while reviewing this confusing code I think it could be made more clear.

Let's rename `err` to `formattingError` and call out explicitly "includeStackTracesInErrorResponses is used in development so it will be helpful to show errors thrown by formatError hooks in that mode". Though honestly maybe we should wrap the error in some sort of "error thrown in formatError hook" error?

Reply:
Implemented your feedback.

> Though honestly maybe we should wrap the error in some sort of "error thrown in formatError hook" error?

Not sure I need to think about it.